File: activity/20200805/junhaolin/1281.Subtract_the_Product_and_Sum_of_Digits_of_an_Integer.cpp
class Solution {
public:
    int subtractProductAndSum(int n) {
        long product = (n == 0 ? 0 : 1);
        long sum = 0;
        while (n) {
            sum += (n % 10);
            product *= (n % 10);
            n /= 10;
        }
        return product - sum;
    }
};

File: activity/20200805/ninezero/1281_subtract_the_product_and_sum_of_digits_of_an_integer.go
func subtractProductAndSum(n int) int {
	product, sum := 1, 0
	length := len(strconv.Itoa(n))
	for i := 0; i < length; i++ {
		product *= n % 10
		sum += n % 10
		n /= 10
	}
	return product - sum
}
